### PR TITLE
Stefan features

### PR DESCRIPTION
**Pull request by S-Ptr**
Added unit tests

### CODE REVIEW

**Review by risticivan12 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
 target/
+<<<<<<< HEAD
```
Comment (aleksandarmarkovic-verisec) [inline .gitignore]: Leave just target here. It seems you had some conflicts in this PR.

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
+/*
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java]: Remove this comments. You can google how to configure Netbeans not to generate you this every time you create new class.

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        try {
+            instance = new SignService();
+        } catch (FrejaEidClientInternalException ex) {
+            Logger.getLogger(SignServiceTest.class.getName()).log(Level.SEVERE, null, ex);
+            fail("Bad init");
+        }
+        System.out.println("Done");
+    }
+    
+    @AfterAll
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java]: Nice to know how this annotations work, but I think we only need before all for now, you can remove rest of them.

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    @AfterEach
+    public void tearDown() {
+        System.out.println("post-test cleanup");
+    }
+
+    /**
+     * Test of initiateSignRequest method, of class SignService.
+     */
+    @Test
+    public void testInitiateSignRequest() throws Exception {
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java:38]: Rename test to describe what it is testing - in this case it is sucessfull case for init sign.
Do this with all test cases.

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    
+    @Test
+    public void testInitiateSignRequestWrongData() throws Exception {
+        System.out.println("initiateSignRequest with irregular data");
+        String email = "asd.324??--dd";
+        String title = "";
+        String dataToSignText = "-------";
+        MinRegistrationLevel registrationLevel = MinRegistrationLevel.BASIC;
+        assertThrows(FrejaEidClientInternalException.class, () -> {instance.initiateSignRequest(email, title, dataToSignText, registrationLevel);});
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java:57]: nice one!

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ *
+ * @author User
+ */
+public class SignServiceTest {
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java:18]: Add tests for getResults

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    public void testInitiateSignRequest() throws Exception {
+        System.out.println("initiateSignRequest with regular data");
+        String email = "aleksandar.markovic@verisec.com";
+        String title = "asd";
+        String dataToSignText = "asd";
+        MinRegistrationLevel registrationLevel = MinRegistrationLevel.BASIC;
+        String expResult = "";
+        String result = instance.initiateSignRequest(email, title, dataToSignText, registrationLevel);
+        System.out.println(result);
+        assertNotEquals("", result);
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java]: maybe its better to assert here that reference is not null

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    @Test
+    public void testInitiateSignRequest() throws Exception {
+        System.out.println("initiateSignRequest with regular data");
+        String email = "aleksandar.markovic@verisec.com";
+        String title = "asd";
+        String dataToSignText = "asd";
+        MinRegistrationLevel registrationLevel = MinRegistrationLevel.BASIC;
+        String expResult = "";
+        String result = instance.initiateSignRequest(email, title, dataToSignText, registrationLevel);
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java:44]: rename this to expectedReference or something more readable

File: nbactions.xml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aleksandarmarkovic-verisec) [inline nbactions.xml:1]: What is this file used for? Why do we need it?

File: src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    
+    @Test
+    public void testInitiateSignRequestEmptyData() throws Exception {
+        System.out.println("Calling initiateSignRequest with empty data");
+        assertThrows(FrejaEidClientInternalException.class, () -> {instance.initiateSignRequest("", "", "", MinRegistrationLevel.BASIC);});
+        System.out.println("PASS");
+    }
+
+
+
```
Comment (aleksandarmarkovic-verisec) [inline src/test/java/com/internship/frejaeidjmeterplugin/jmeter/frejaRequests/SignServiceTest.java:69]: Format code

**Pull request closed without merging**